e2e/vmss.go

```diff
@@ -82,55 +80,46 @@ func createVMSSWithPayload(ctx context.Context, customData, cseCmd, vmssName str
 		return nil, fmt.Errorf(" prepare model for VMSS %q: %w", vmssName, err)
 	}
 
-	var pollErr error
-	var vmssResp *armcompute.VirtualMachineScaleSetsClientCreateOrUpdateResponse
-
-	for i := 0; i < maxRetries; i++ {
-		createVMSSCtx, cancel := context.WithTimeout(ctx, vmssClientCreateVMSSPollingTimeout)
-		defer cancel()
-
-		vmssResp, pollErr = pollVMSSOperation(createVMSSCtx, vmssName, pollVMSSOperationOpts{
-			pollUntilDone: &runtime.PollUntilDoneOptions{
-				Frequency: vmssClientCreateVMSSPollInterval,
-			},
-		},
-			func() (Poller[armcompute.VirtualMachineScaleSetsClientCreateOrUpdateResponse], error) {
-				return config.Azure.VMSS.BeginCreateOrUpdate(
-					ctx,
-					*opts.clusterConfig.cluster.Properties.NodeResourceGroup,
-					vmssName,
-					model,
-					nil,
-				)
-			})
-
-		if pollErr == nil {
-			return &vmssResp.VirtualMachineScaleSet, nil
-		}
-		if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
-			return nil, fmt.Errorf("create VMSS %q: %w", vmssName, pollErr)
-		}
-		log.Printf("failed to create VMSS. Retry attempts left: %d", maxRetries-(i+1))
+	operation, err := config.Azure.VMSS.BeginCreateOrUpdate(
+		ctx,
+		*opts.clusterConfig.Model.Properties.NodeResourceGroup,
+		vmssName,
+		model,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("begin create VMSS %q: %w", vmssName, err)
 	}
-	return nil, fmt.Errorf("create VMSS %q: %w", vmssName, pollErr)
+	vmssResp, err := operation.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
+		Frequency: 10 * time.Second,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("poll until done for VMSS %q: %w", vmssName, err)
+	}
+	return &vmssResp.VirtualMachineScaleSet, nil
 }
 
 // Adds additional IP configs to the passed in vmss model based on the chosen cluster's setting of "maxPodsPerNode",
 // as we need be able to allow AKS to allocate an additional IP config for each pod running on the given node.
 // Additional info: https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni
-func addPodIPConfigsForAzureCNI(vmss *armcompute.VirtualMachineScaleSet, vmssName string, opts *scenarioRunOpts) error {
-	maxPodsPerNode, err := opts.clusterConfig.maxPodsPerNode()
+func addPodIPConfigsForAzureCNI(ctx context.Context, vmss *armcompute.VirtualMachineScaleSet, vmssName string, opts *scenarioRunOpts) error {
+	maxPodsPerNode, err := opts.clusterConfig.MaxPodsPerNode()
 	if err != nil {
 		return fmt.Errorf("failed to read agentpool MaxPods value from chosen cluster model: %w", err)
 	}
 
+	vnet, err := getClusterVNet(ctx, *opts.clusterConfig.Model.Properties.NodeResourceGroup)
```

Comment:
could this also be something we instantiate on the cluster object before running any scenarios?

Reply:
especially since we also use it in `getBaseVMSSModel`